### PR TITLE
Add non-functional requirements and 'edit tag' use case to DeveloperGuide.md

### LINKED ISSUE #9: Add our project's non-functional requirements to the Developer Guide

### PR DESCRIPTION
**Pull request by khinkhinn**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -367,6 +367,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 2a. The list is empty.
+     
+     Use case ends.
+     
+* 3a. The given index is invalid.
+
+    * 3a1. Projact shows an error message.
+
+      Use case resumes at step 2.
+
```
Comment (yeo-kexin) [inline docs/DeveloperGuide.md:398]: should we include a case for when the given tag name is also invalid? (e.g. t/ is missing)

**Comment by akgrenSoar** [comment]
Closes #9

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -367,6 +367,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
    
    Use case ends.
   
+**Use case: Edit tag*
```
Comment (akgrenSoar) [inline docs/DeveloperGuide.md]:
```suggestion
**Use case: Edit tag**
```
Formatting

**Comment by akgrenSoar** [comment]
Closes #7 , Closes #9

**Review by yeo-kexin — APPROVED** [review]

**Pull request merged**